lib/generate/browserCollector.js

```diff
@@ -0,0 +1,191 @@
+                     * This make it possible to fetch real module paths even with bundling enabled.
+                     */
+                    const contexts = require.s.contexts;
+                    const defContext = contexts._;
+                    const defaultContextConfig = defContext.config;
+                    const unbundledContextConfig = {
+                        baseUrl: defaultContextConfig.baseUrl,
+                        paths: defaultContextConfig.paths,
+                        shim: defaultContextConfig.shim,
+                        config: defaultContextConfig.config,
+                        map: defaultContextConfig.map,
+                    };
+                    const unbundledContext = require.s.newContext('magepack');
+                    unbundledContext.configure(unbundledContextConfig);
+
+                    require.onResourceLoad = function (context, map, depArray) {
+                        // Modules are defined as being loaded 5 seconds after the
+                        // final resourceLoad event is fired. Immediately set the
+                        // loaded state to false and then set up a timer.
+                        window.modulesLoaded = false;
+                        if (window.modulesLoadedTimer) {
+                            // Delete the existing timer
+                            window.clearTimeout(window.modulesLoadedTimer);
+                        }
+                        window.modulesLoadedTimer = window.setTimeout(() => {
+                            window.modulesLoaded = true;
+                        }, 5000);
+                        // TODO: is there a queue we can check instead? This seems
+                        // too hackish to be consistent on all hardware / networks
+                        // e.g.: checkout, are we still waiting on estimate shipping
+
+                        // build up proper normalized moduleID dependency tree
+                        var moduleName = map.unnormalized
+                            ? map.prefix + '!' + map.name
+                            : map.id;
+
+                        if (excludedModules.includes(moduleName)) {
+                            return;
+                        }
+
+                        // Ignore all modules that are loaded with plugins other than text.
+                        if (
+                            moduleName.includes('!') &&
+                            !moduleName.startsWith('text!')
+                        ) {
+                            return;
+                        }
+
+                        // Ignore all modules loaded by URL
+                        if (moduleName.match(/^(https?:)?\/\//)) {
+                            return;
+                        }
+
+                        var deps = [];
+                        for (var i = 0; i < depArray.length; i++) {
+                            let depName;
+                            if (depArray[i].unnormalized) {
```

Comment:
Consider moving this (lines 112-131) to its own function to reduce the duplicate code with the above (lines 87-107). For example:

```javascript
const getModuleName = (module, allowPlugins) => {
    let moduleName;

    if (module.unnormalized) {
        // Plugin loading - normalise the dependency name
        moduleName = module.prefix + '!' + module.name;
    } else {
        moduleName = module.id;
    }

    // Ignore all modules loaded by URL
    if (moduleName.match(/^(https?:)?\/\//)) {
        return;
    }

    // Ignore all modules that are loaded with plugins other than text.
    if (moduleName.includes('!') && !moduleName.startsWith('text!')) {
        if (allowPlugins !== true) {
            return;
        }

        moduleName = stripPlugin(moduleName);
    }

    if (excludedModules.includes(moduleName)) {
        return;
    }

    return moduleName;
};

// used like:
const moduleName = getModuleName(map);
if (!moduleName) {
    return;
}

// and:
const depName = getModuleName(depArray[i], true);
if (depName) {
    deps.push(depName);
}
```